src/couch/src/couch_file.erl

```diff
@@ -596,7 +604,28 @@ handle_call({write_header, Bin}, _From, #file{fd = Fd, eof = Pos} = File) ->
             {reply, Error, reset_eof(File)}
     end;
 handle_call(find_header, _From, #file{fd = Fd, eof = Pos} = File) ->
-    {reply, find_header(Fd, Pos div ?SIZE_BLOCK), File}.
+    {reply, find_header(Fd, Pos div ?SIZE_BLOCK), File};
+handle_call({commit_header, Bin}, From, #file{fd = Fd} = File) ->
+    case fsync(Fd) of
+        ok ->
+            Result = handle_call({write_header, Bin}, From, File),
```

Comment:
I find this quite ugly, can we extract the common code into a function that both places call please?

Reply:
addressed in 050ae0742a10